### ISSUE TITLE
Type-aware rules crash with "type handle not found in snapshot registry" on files using Partial<T> / Record<K, V>

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#492]
### Describe the bug

When `oxlint-plugin-awscdk` runs on TypeScript files that use generic utility types such as `Partial<T>` or `Record<K, V>` in interface members, the plugin aborts with the following error:

```text
Error running JS plugin. File path: <file>
Error: protocol error: api: client error: type handle "synthetic-type-argument:<id>:<index>:<TypeName>" not found in snapshot registry
    at CorsaProjectSession.getSymbolOfType (.../corsa-oxlint/dist/session.js:116:50)
    at Object.getSymbolOfType (.../corsa-oxlint/dist/checker.js:49:46)
    at isExtendsFromTargetSuperClass (.../oxlint-plugin-awscdk/dist/index.mjs:88:23)
    at isResourceType (.../oxlint-plugin-awscdk/dist/index.mjs:225:9)
    at isResourceWithReadonlyInterface (.../oxlint-plugin-awscdk/dist/index.mjs:241:16)
    at findTypeOfCdkConstruct (.../oxlint-plugin-awscdk/dist/index.mjs:287:6)
    at findFromTypeArguments (.../oxlint-plugin-awscdk/dist/index.mjs:307:21)
    ...
```

The lint run exits with a non-zero status (no other rule violations are reported for the file once the plugin crashes).

Reproduced on both `oxlint-plugin-awscdk@0.1.1` and `@0.1.2`.

### Expected Behavior

It would be helpful if `oxlint-plugin-awscdk` could analyze files that contain generic utility types (`Partial<T>`, `Record<K, V>` etc.) without crashing — for example by gracefully skipping resolution when a synthetic type argument is not present in the snapshot registry.

### Current Behavior

The plugin throws the `synthetic-type-argument:...:<TypeName>` error and aborts before any rule is evaluated for the affected file.

**Cases observed (extracted to minimal patterns):**

| Case | `<TypeName>` in crash | Triggering generic usage |
| --- | --- | --- |
| 1 | `AlarmProps` | `readonly alarmProps?: Partial<AlarmProps>` from `aws-cdk-lib/aws-cloudwatch` |
| 2 | `string` | `readonly someMap?: Record<string, string>` |

**Minimal reproduction — case 1 (`Partial<T>`):**

```ts
import { AlarmProps } from 'aws-cdk-lib/aws-cloudwatch';
import { Construct } from 'constructs';

export interface FactoryProps {
  readonly alarmProps?: Partial<AlarmProps>;
}

export class Factory extends Construct {
  constructor(scope: Construct, id: string, _props: FactoryProps) {
    super(scope, id);
  }
}
```

**Minimal reproduction — case 2 (`Record<K, V>`):**

```ts
import { Construct } from 'constructs';

export interface FactoryProps {
  readonly someMap?: Record<string, string>;
}

export class Factory extends Construct {
  constructor(scope: Construct, id: string, _props: FactoryProps) {
    super(scope, id);
  }
}
```

**Steps to reproduce:**

1. Place either of the files above in a TypeScript CDK project.
2. Configure `oxlint-plugin-awscdk` via `oxlint` (`jsPlugins: ['oxlint-plugin-awscdk']`) or via `vite-plus` (`vp lint`).
3. Run the lint command. The plugin crashes for the file with the stack trace above.

### Plugin

oxlint-plugin-awscdk

### Plugin Version

0.1.2

### Language Version

TypeScript (5.8.3)

### Additional Context

**Environment (machine-readable):**

```yaml
oxlint-plugin-awscdk: 0.1.2  # also reproducible on 0.1.1
corsa-oxlint:        0.49.0
"@oxlint/plugins":   1.68.0
oxlint:              1.70.0
oxlint-tsgolint:     0.23.0
vite-plus:           0.2.1
typescript:          5.8.3
node:                v24.18.0
pnpm:                10.8.1
aws-cdk-lib:         2.222.0
constructs:          10.3.0
os:                  macOS (darwin 23.1.0)
```

**Stack trace — case 1 (`Partial<AlarmProps>`):**

```text
<file>: error: Error running JS plugin.
File path: <file>
Error: protocol error: api: client error: type handle "synthetic-type-argument:<id>:0:AlarmProps" not found in snapshot registry
    at CorsaProjectSession.getSymbolOfType (.../corsa-oxlint@0.49.0/.../session.js:116:50)
    at Object.getSymbolOfType (.../corsa-oxlint@0.49.0/.../checker.js:49:46)
    at isExtendsFromTargetSuperClass (.../oxlint-plugin-awscdk@0.1.2/.../index.mjs:88:23)
    at isResourceType (.../oxlint-plugin-awscdk@0.1.2/.../index.mjs:225:9)
    at isResourceWithReadonlyInterface (.../oxlint-plugin-awscdk@0.1.2/.../index.mjs:241:16)
    at findTypeOfCdkConstruct (.../oxlint-plugin-awscdk@0.1.2/.../index.mjs:287:6)
    at findFromTypeArguments (.../oxlint-plugin-awscdk@0.1.2/.../index.mjs:307:21)
    at findTypeOfCdkConstruct (.../oxlint-plugin-awscdk@0.1.2/.../index.mjs:288:55)
    at findFromUnionOrIntersection (.../oxlint-plugin-awscdk@0.1.2/.../index.mjs:297:21)
    at findTypeOfCdkConstruct (.../oxlint-plugin-awscdk@0.1.2/.../index.mjs:288:9)
```

**Stack trace — case 2 (`Record<string, string>`):**

```text
<file>: error: Error running JS plugin.
File path: <file>
Error: protocol error: api: client error: type handle "synthetic-type-argument:<id>:0:string" not found in snapshot registry
    at CorsaProjectSession.getSymbolOfType (.../corsa-oxlint@0.49.0/.../session.js:116:50)
    at Object.getSymbolOfType (.../corsa-oxlint@0.49.0/.../checker.js:49:46)
    at isExtendsFromTargetSuperClass (.../oxlint-plugin-awscdk@0.1.2/.../index.mjs:88:23)
    ... (same frames as case 1)
```

**Hypothesis:**

`findFromTypeArguments` / `isExtendsFromTargetSuperClass` walks the type arguments of generic types (`Partial<T>`, `Record<K, V>` etc.) and tries to resolve them via `CorsaProjectSession.getSymbolOfType`. The internal "snapshot registry" of `corsa-oxlint` appears to not contain a symbol for synthetic type arguments produced by these utility types, which causes the crash.

**Related upstream:**

The error originates from `corsa-oxlint`'s `CorsaProjectSession.getSymbolOfType`. If the snapshot registry behavior is owned by `corsa-oxlint` rather than this plugin, it may be worth coordinating with the upstream maintainers as well — happy to file a follow-up issue there if that helps.

**Editor / Tooling:**

- Runner: `oxlint` directly, or `vp lint` (vite-plus)
- Package manager: pnpm
